tests/test_pyrflx.py

```diff
@@ -928,18 +930,29 @@ def test_ethernet_generating_ethernet_2_vlan(self) -> None:
             msg_as_bytes: bytes = file.read()
         self.assertEqual(self.frame.bytestring, msg_as_bytes)
 
-    # ISSUE: Componolit/RecordFlux#199
-
-    # def test_ethernet_generating_ethernet_2_vlan_dynamic(self) -> None:
-    #    raise NotImplementedError
+    # test is not relevant for the python implementation
+    def test_ethernet_generating_ethernet_2_vlan_dynamic(self) -> None:
+        pass
 
     # rflx in_ipv4-tests.adb
 
-    # def test_ipv4_parsing_udp_in_ipv4(self) -> None:
-    #    raise NotImplementedError
+    def test_ipv4_parsing_udp_in_ipv4(self) -> None:
+        with open("tests/ipv4_udp.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
 
-    # def test_ipv4_parsing_udp_in_ipv4_in_ethernet(self) -> None:
-    #    raise NotImplementedError
+        self.ipv4.parse(msg_as_bytes)
+        nested_udp = self.ipv4.get("Payload")
+        self.assertTrue(nested_udp.valid_message)
+
+    def test_ipv4_parsing_udp_in_ipv4_in_ethernet(self) -> None:
+        with open("tests/ethernet_ipv4_udp.raw", "rb") as file:
+            msg_as_bytes: bytes = file.read()
+
+        self.frame.parse(msg_as_bytes)
+        nested_ipv4 = self.frame.get("Payload")
+        self.assertTrue(nested_ipv4.valid_message)
+        nested_udp = nested_ipv4.get("Payload")
+        self.assertTrue(nested_udp.valid_message)
```

Comment:
I think that would be a good place to check the newly introduced `identifier`, `name` and `package` methods (at least `package` is not covered by any unit tests).

Reply:
Added tests in bfda76d. Are these enough or did you mean to test package, identifier and name for all messages in this unit test?

Reply:
Current code:

```Python
        self.assertEqual(str(nested_udp.package.name), "UDP")
        self.assertEqual(str(nested_udp.identifier.name), "Datagram")
```

You should check the identifier directly, not just its name:

```Python
        self.assertEqual(nested_udp.identifer, ID("UDP.Datagram"))
```

I think it makes sense to check the same for `nested_ipv4`.

`package` and `name` should be also tested. You could do that here (as shown below) or in a separate unit test. (The latter would be the cleaner solution.)

```Python
        self.assertEqual(nested_udp.package, ID("UDP"))
        self.assertEqual(nested_udp.name, "Datagram")
```

Reply:
Added a separate test for name and package in b3ec99d